adr-notification-system.md

```diff
@@ -0,0 +1,33 @@
+# ADR 2: Extract Notification Component
+
+## Status
+
+PROPOSED
+
+## Context
+
+Notification mechanisms are used to notify users and experts about changes in the state of a ticket workflow and to send surveys. Currently, it's implemented as a component in the monolith system and uses external sms and email gateways.
+
+## Decision
+
+1. Move notification component into a separate module.
+
+2. The module should use no underlying data source.
+
+3. The main concern is, provided with all the necessary data to generate notifications, to handle the communication with external gateways, and handle delays, timeouts and exceptions gracefully.
+
+4. Internal modules should use the notification module in a "fire-and-forget" fashion.
```

Comment:
Again, I think this is more of a design principle than a decision as is, but if we highlighted the fact that we just want to throw notification tasks into a queuing system, it will probably make a bit more sense here:
```suggestion
3. The module should receive tasks via a messaging system, hence modules will be able to use the notification module in a "fire-and-forget" fashion.
```